File: Assets/scenes/warzone-landing/script/ExplodingBaseBehaviour.js
#pragma strict

/* base that crumbles and explodes */

public var rumbleSound : AudioClip;

public var bigExplosion : BigExplosionBehaviour;

public var rockParts : Transform[];
public var rockVelocity : Vector3[];

public var exploding : boolean = false;


public var test : boolean = false;

function Start () {
	gameObject.GetComponent.<Light>().intensity = 0.0f;
	
}

function FixedUpdate () {
	if(test){
		test = false;
		if(!exploding){
			startFallingApart();
		} else {
			finalExplosion();
		}
	}
	if(exploding){
		for (var t = 0; t < rockVelocity.length; t++){
			rockParts[t].position += rockVelocity[t];
		}
	}
}


function startFallingApart(){
	UsefulShit.PlayClipAt(rumbleSound, transform.position);
	exploding = true;
	gameObject.GetComponent.<Light>().intensity = 8.0f;
	
	for(var ps : ParticleSystem in GetComponentsInChildren.<ParticleSystem>()){
		ps.enableEmission = true;
	}
	
	//start a particle effect too
}

function finalExplosion(){
	
	bigExplosion.Explode();
	for (var t = 0; t < rockVelocity.length; t++){
		rockVelocity[t] *= 10.0f;
	}
}

File: Assets/scenes/warzonescene/script/ExplodingBaseBehaviour.cs
using UnityEngine;
using System;


public class ExplodingBaseBehaviour:MonoBehaviour{
	
	/* base that crumbles and explodes */
	
	public AudioClip rumbleSound;
	public AudioClip finalExplosionSound;
	
	public BigExplosionBehaviour bigExplosion;
	
	public Transform[] rockParts;
	public Vector3[] rockVelocity;
	
	public bool exploding = false;
	public float explosionStartTime;
	
	
	public bool test = false;
	
	public void Start() {
		gameObject.GetComponent<Light>().intensity = 0.0f;
		
	}
	
	public void FixedUpdate() {
		if(test){
			test = false;
			if(!exploding){
				startFallingApart();
			} else {
				finalExplosion();
			}
		}
		if(exploding){
			if(Time.fixedTime - explosionStartTime < 90.0f){		//only separate the parts of the base for the first 90 seconds of the explosion
				for(int t = 0; t < rockVelocity.Length; t++){
					rockParts[t].position += rockVelocity[t];
				}
			}
		}
	}
	
	
	public void startFallingApart(){
		UsefulShit.PlayClipAt(rumbleSound, transform.position);
		exploding = true;
		gameObject.GetComponent<Light>().intensity = 8.0f;
		explosionStartTime = Time.fixedTime;
		foreach(ParticleSystem ps in GetComponentsInChildren<ParticleSystem>()){
			ps.enableEmission = true;
		}
		
		//start a particle effect too
	}
	
	public void finalExplosion(){
		AudioSource.PlayClipAtPoint(finalExplosionSound, transform.position);
		bigExplosion.Explode();
		for(int t = 0; t < rockVelocity.Length; t++){
			rockVelocity[t] *= 10.0f;
		}
	}
}